src/components/PersonLink.tsx

```diff
@@ -0,0 +1,56 @@
+import { Link, useLocation, useParams } from 'react-router-dom';
+import { Person } from '../types';
+import classNames from 'classnames';
+
+type Props = {
+  person: Person;
+  findParentPerson: (parentName: string) => Person | undefined;
+};
+
+const getLinkClass = (person: Person) => {
+  return classNames({ 'has-text-danger': person.sex === 'f' });
+};
+
+export const PersonLink: React.FC<Props> = ({ person, findParentPerson }) => {
+  const { slug } = useParams();
+  const location = useLocation();
+
+  const renderParentLink = (parentName: string | null) => {
+    const parent = parentName ? findParentPerson(parentName) : undefined;
+
+    return parent ? (
+      <Link
+        className={getLinkClass(parent)}
+        to={`/people/${parent.slug}${location.search}`}
+      >
+        {parent.name}
+      </Link>
+    ) : (
+      parentName || '-'
+    );
```

Comment:
You are using the ternary operator to check if parent exists and return a Link component or a string respectively. This is a good approach, but you can make it cleaner by returning early when parent does not exist. For example: `if (!parent) { return parentName || '-'; }` and then return the Link component. This way, you avoid the ternary operator and make your code more readable.